src/routes/files.ts

```diff
@@ -22,17 +22,41 @@ export default async function files(fastify: FastifyInstance) {
 
   fastify.get('/files/credentials', async (req) => {
     const sessionCache: SessionCache = req.sessionCache;
+    const results = new Map<ContactType, String[][]>();
     const places = sessionCache.getPlaces();
-    const refinedRecords = transform(places, (place: Place) => {
-      return [
-        place.type.friendly,
+    places.forEach(place => {
+      const parent = Config.getParentProperty(place.type);
+      const record = [
+        place.hierarchyProperties[parent.property_name],
         place.name,
         place.creationDetails.username,
         place.creationDetails.password,
         place.creationDetails.disabledUsers,
       ];
+      const result = results.get(place.type) || [];
```

Comment:
can't we move this to line 28? do it once outside the loop

Reply:
It has to be inside the loop, we're building the rows for each place's type

Reply:
my mistake. it is the repetitive call to `set` that isn't needed. 
is the Map required? did you consider just using an object? believe you can do it in one line

something like: 
```
const results: { [key: string]: Array<string> } = {};
results[place.type] = [...results[place.type], record];
```

Reply:
I'd be using the object as a Map so i just used Map.

`results[place.type] = [...results[place.type], record];`

I think we'd be copying the array each time so while short, it's likely it's less efficient

Reply:
i believe spread operator performs faster in benchmarks ([example](https://stackoverflow.com/questions/39716216/difference-between-using-a-spread-syntax-and-push-apply-when-dealing-with))

the spread operator isn't a copy... and most of this is triviall optimized by a compiler 
overall, you should optimize for human brain space. that means prioritising readable code as the top priority rather than over optimizing small things